enhancing/enhancer.js

```diff
@@ -6,17 +6,37 @@ module.exports = {
 };
 
 function succeed(item) {
-  return { ...item };
+  const { enhancement } = item;
+  if(enhancement < 20) {
+    item.enhancement += 1;
+  }
+  return { ...item};
 }
 
 function fail(item) {
+  const { enhancement } = item;
+  if(enhancement > 16) {
+    item.durability -= 10, item.enhancement -= 1;
```

Comment:
```suggestion
    item.durability -= 10; 
    item.enhancement -= 1;
```
Since this is in a block wouldn't you want it to be like this?